File: Back-End/php/dblib/ValueObjects/Game.php
<?php
/**
 * Projeto FutMesaBackEnd
 *
 * @copyright : Renato Martins Barbieri Nunes
 * @version 0.1
 */
namespace ValueObject;

require_once "ValueObjects/ValueObject.php";

/**
 * Tabela de jogos de campeonatos do banco de dados.
 */
class Game extends ValueObject
{

   /**
    *
    * @var int Identificador do jogo.
    */
   public $id;

   /**
    *
    * @var int Identificador da rodada.
    */
   public $idround;

   /**
    *
    * @var int Identificador do jogador 1 do jogo.
    */
   public $idplayer1;

   /**
    *
    * @var int Identificador do jogador 2 do jogo.
    */
   public $idplayer2;

   /**
    *
    * @var int Número da mesa de jogo.
    */
   public $gametable;

   /**
    *
    * @var int Gols do jogador 1 no jogo.
    */
   public $score1;

   /**
    *
    * @var int Gols do jogador 2 no jogo.
    */
   public $score2;

   /**
    *
    * @var mixed Horário que o resultado foi inserido.
    */
   public $inputdate;

   /**
    *
    * @var int Identificador do jogador vencedor.
    */
   public $idwinner;
   
   /**
    * Preenche as informações deste objeto a partir de outro do mesmo tipo.
    *
    * @param Game $game
    */
   public function copyData( Game $game ) : void
   {
      $this->id = $game->id;
      $this->idround = $game->idround;
      $this->idplayer1 = $game->idplayer1;
      $this->idplayer2 = $game->idplayer2;
      $this->gametable = $game->gametable;
      $this->score1 = $game->score1;
      $this->score2 = $game->score2;
      $this->inputdate = $game->inputdate;
      $this->idwinner = $game->idwinner;
   }
}

File: Front-End/GWT/src/com/futmesa/client/businessinteligence/Game.java
package com.futmesa.client.businessinteligence;

import com.google.gwt.core.client.JavaScriptObject;
import com.google.gwt.view.client.ProvidesKey;

/**
 * Classe para representar um jogo de um campeonato retornado pelo back-end.
 */
public final class Game extends JavaScriptObject
{
   
   /**
    * Construtor que não será chamado.
    */
   protected Game()
   {
      // Não faz nada mesmo.
   }

   /**
    * The key provider that provides the unique ID of a contact.
    */
   public static final ProvidesKey<Game> KEY_PROVIDER = new ProvidesKey<Game>() {
     @Override
     public Object getKey(Game item) {
       return item == null ? null : item.getId();
     }
   };
   
   /**
    * @return Retorna o identificador do jogo.
    */
   public final native int getId() /*-{
      return this.id;
   }-*/;

   /**
    * @return Retorna o identificador da rodada do campeonato.
    */
   public final native int getIdRound() /*-{
      return this.idround;
   }-*/;

   /**
    * @return Retorna o identificador do jogador 1.
    */
   public final native int getPlayer1Id() /*-{
      return this.player1.id;
   }-*/;

   /**
    * @return Retorna o nome do jogador 1.
    */
   public final native String getPlayer1Name() /*-{
      return this.player1.name;
   }-*/;
   
   /**
    * @return Retorna o identificador do jogador 2.
    */
   public final native int getPlayer2Id() /*-{
      return this.player2.id;
   }-*/;

   /**
    * @return Retorna o nome do jogador 2.
    */
   public final native String getPlayer2Name() /*-{
      return this.player2.name;
   }-*/;
   
  /**
   * @return Mesa do jogo.
   */
  public final native int getGameTable() /*-{
  	 return this.gametable;
  }-*/;

  /**
   * @return Gols do jogador 1 no jogo.
   */
  public final native int getScore1() /*-{
  	 return this.score1;
  }-*/;

  /**
   * @return Gols do jogador 2 no jogo.
   */
  public final native int getScore2() /*-{
  	 return this.score2;
  }-*/;

  /**
   * @return Horário que o resultado foi inserido.
   */
  public final native String getInputDate() /*-{
  	 return this.inputdate;
  }-*/;
  
  /**
   * @return Identificador do jogador vencedor.
   */
  public final native String getIdWinner() /*-{
  	 return this.idwinner;
  }-*/;

	public final native void setScore1(int score) /*-{
		this.score1 = score;
	}-*/;

	public final native void setScore2(int score) /*-{
		this.score2 = score;
	}-*/;

	public final native void setPlayer1Name(String name) /*-{
		  if (!this.player1) {
		  	this.player1 = {};
		  }
	      this.player1.name = name;
	   }-*/;

	public final native void setPlayer2Name(String name) /*-{
		if (!this.player2) {
		  	this.player2 = {};
		  }
	      this.player2.name = name;
	   }-*/;

}

